app/javascript/components/users/authentication/forms/SigninForm.jsx

```diff
@@ -38,10 +38,24 @@ export default function SigninForm() {
   const captchaRef = useRef(null);
   const { data: env } = useEnv();
 
+  // get history and location objects
+  const location = useLocation();
+  const navigate = useNavigate();
+
   const handleSubmit = useCallback(async (session) => {
     const results = await captchaRef.current?.execute({ async: true });
     const token = results?.response || '';
 
+    // need to check if theres a location param
+    const roomUrl = new URLSearchParams(location.search).get('location');
+    if (roomUrl) {
+      // Redirect the user to the joinCard for the room
+      navigate(roomUrl);
+    } else {
+      // Redirect the user to the rooms homepage
+      navigate('/rooms');
+    }
+
     return createSessionAPI.mutate({ session, token });
```

Comment:
Look into the `useCreateSession` mutation - that's where the current redirect happens